src/sentry/static/sentry/app/components/stream/group.tsx

```diff
@@ -61,37 +84,28 @@ const StreamGroup = createReactClass({
     selection: SentryTypes.GlobalSelection.isRequired,
     organization: SentryTypes.Organization.isRequired,
     useFilteredStats: PropTypes.bool,
-  },
-
-  mixins: [Reflux.listenTo(GroupStore, 'onGroupChange')],
+  };
 
-  getDefaultProps() {
-    return {
-      id: '',
-      statsPeriod: '24h',
-      canSelect: true,
-      withChart: true,
-      useFilteredStats: false,
-    };
-  },
+  static defaultProps = defaultProps;
 
-  getInitialState() {
-    const data = GroupStore.get(this.props.id);
+  constructor(props: Props) {
+    super(props);
+    const data = GroupStore.get(props.id) as Group;
 
-    return {
+    this.state = {
```

Comment:
should we move it out of the constructor? it's just a suggestion ;-)

Reply:
I didn't like how it looked with GroupStore needing to be called multiple times

Reply:
I see...

I believe it's fine if you do something like:

```
  state: State = this.getInitialState();

  getInitialState() {
    const {id, useFilteredStats} = this.props;

    const data = GroupStore.get(id) as Group;

    return {
      data: {
        ...data,
        filtered: useFilteredStats ? data.filtered : undefined,
      },
    };
  }
```